src/DynamoApplications/StartupUtils.cs

```diff
@@ -0,0 +1,302 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.Interfaces;
+using DynamoShapeManager;
+using System.Reflection;
+using System.IO;
+using Dynamo.Models;
+using Dynamo.UpdateManager;
+using Microsoft.Win32;
+using System.Diagnostics;
+using System.Threading;
+using System.Globalization;
+using NDesk.Options;
+
+namespace Dynamo.Applications.StartupUtils
+{
+    internal class PathResolver : IPathResolver
+    {
+        private readonly List<string> additionalResolutionPaths;
+        private readonly List<string> additionalNodeDirectories;
+        private readonly List<string> preloadedLibraryPaths;
+
+        public PathResolver(string preloaderLocation, IEnumerable<string> librariesToPreload)
```

Comment:
I've modified the signature for PathResolver to take a ienumerable<string> or libraries so we can pass the specific libraries to load instead of implementing a new path resolver subclass